src/parse.rs

```diff
@@ -2228,6 +2230,48 @@ struct LinkDef<'a> {
     title: Option<CowStr<'a>>,
 }
 
+struct CodeDelims {
+    inner: HashMap<usize, VecDeque<TreeIndex>>,
```

Comment:
I think a comment explaining exactly what this represents would be useful.

Another thing to keep in mind is that an empty HashMap still has an allocation (one reason I sometimes prefer BTreeMap for similar purposes). But since this seems to be once per parse, shouldn't be a problem.

Reply:
TIL - I did not know about empty hashmaps allocating. When hashbrown is merged to the standard library this should no longer be the case I believe.

Reply:
Good to know. Looks like hashbrown is going to be a major improvement!